Bug Fixes and Changelog.md

nmn opened the pull request:
- Throttle .flowconfig errors to once every 5 minutes
- revert back to simple @flow detecting to fix regressions
- More cases where we need to ignore flow cli errors and keep trying
- Add Changelog.md starting with v5.0.0

nmn commented:
@steelbrain @Adezandee 
This should fix many of the newly created issues.

nmn commented:
I can get this merged and released if I get a code-review.

Arcanemagus commented on lib/index.js:
```diff
@@ -49,7 +49,7 @@ export default {
   deactivate(): void {
     const helpers = require('atom-linter');
-    console.log('deactivating... lintet-flow');
+    console.log('deactivating... linter-flow');
```
Might wrap this in an `is (atom.inDevMode())` check?

Arcanemagus commented on lib/index.js:
```diff
@@ -73,21 +73,20 @@ export default {
-        if (firstComStart !== -1 &&
-          firstComEnd !== -1 &&
-          fileText.slice(firstComStart + 2, firstComEnd).indexOf('@flow') === -1) {
+        if (!fileText || fileText.indexOf('@flow') === -1) {
```
Nice catch of an empty file :+1:

Arcanemagus commented on lib/index.js:
```diff
@@ -73,21 +73,20 @@ export default {
-            detail: 'To get started with Flow, run `flow init`.',
-            dismissable: true,
-          });
+          if (!this.lastConfigError || this.lastConfigError + 5 * 60 * 1000 < Date.now()) {
```
Maybe tie this into the filepath as well? So that it only warns every x minutes for each file, but if you switch to a new file immediately it will warn again. I'm not sure how useful this would be though since the entire point of this change is to make that less noisy.

You might log to the console every time, regardless of the timer though.

nmn replied:
wrap this console.log in an `is (atom.inDevMode())` check as well?

Arcanemagus replied:
Well since it would only be logging when there is an issue, and the user would only see this if they open Developer Tools I would probably leave it in.

Generally the idea from what I've seen is leave the console clean unless there is something wrong, or the user is running Atom in dev mode (and thus probably wants to see as much information as possible).

nmn commented:
@Arcanemagus Addressed the comments. Would love some quick code reviews and get this merged. People are having trouble and this should fix most of it.

Arcanemagus commented:
@nmn Review of code is important, but remember that if you have a verified fix ready you should probably just get it out there as there may not be somebody around to review things in a timely manner on a small project like this :wink:

Thanks for taking care of this!

nmn commented:
Let me know if more bugs pop up.
